### PR TITLE
fix/import-nfts

### PR DESCRIPTION
**Pull request by leozera14**
- [x] Implement function to handle ERC721 NFTs
- [x] Implement supportsInterface function to handle ERC1155 error and verify which type we are receiving
- [ ] Test supportsInterface function and fix it if needs

### CODE REVIEW

File: packages/sysweb3-utils/src/getContract.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  const provider = new Web3.providers.HttpProvider(networkUrl);
+
+  const web3Provider = new Web3();
+
+  web3Provider.setProvider(provider);
+
+  const [abi721, abi1155] = [getErc21Abi(), getErc55Abi()];
+
+  const abisInfos = [
+    { abi: abi721, interfaceId: '0x80ac58cd', type: 'ERC-721' },
```
Comment (claudiovb) [inline packages/sysweb3-utils/src/getContract.ts:18]: Additionally to the base extension erc721 extensions interface as well:
URIStorage: 0xc87b56dd
Enumerable: 0x79f154c4
Burnable: 0x42966c68
Royalty: 0x01ffc9a7

File: packages/sysweb3-utils/src/getContract.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+) => {
+  const provider = new Web3.providers.HttpProvider(networkUrl);
+
+  const web3Provider = new Web3();
+
+  web3Provider.setProvider(provider);
+
+  const [abi721, abi1155] = [getErc21Abi(), getErc55Abi()];
+
+  const abisInfos = [
```
Comment (claudiovb) [inline packages/sysweb3-utils/src/getContract.ts:17]: Add the base interface for erc-20:
Erc20 base interface: 0x36372b07

**Review by claudiovb — CHANGES_REQUESTED** [review]
Besides the comments below. 
Testing should be made for each new exported function created so we can easily check the behavior of the given function.
getERC721StandardBalance --> must have testing trying to call the balance of an ERC721 token at least on syscoin and Mumbai. Also trying to call the balance of an ERC20 and an ERC1155 just for check purposes of the behavior.
contractChecker -> Needs to be refactored, the input arguments are correct but the desired behavior is that this functions returns if the given address  is a contract with the following possibilities: [1 - undefined Contract (is a contract but supportsInterface either returns false or is a nonexistent function), 2 - ERC721, 3 - ERC20, 4 - ERC11555 ]

In order to help the tests below there is some example of contract address that can be queried for testing:
Syscoin:
PSYS (ERC-20) : 0xE18c200A70908c89fFA18C628fE1B83aC0065EA4
PSYSLP (ERC-20): 0x2CDF912CbeaF76d67feaDC994D889c2F4442b300
USDC (ERC-20): 0x2bF9b864cdc97b08B6D79ad4663e71B8aB65c45c
SYSPUNKS (ERC-721): 0x806AC3D3a9EEbC5c09553b5b406730b6FdBb1B23/transactions
WebDomains (ERC-721): 0xA1019535E6b364523949EaF45F4B17521c1cb074
SUSNFT (ERC-1155): 0x7AD299a564E3EC666dB683F632F5a6aD09084eFd
LuxyLaunchpadFeeManager (undefined contract): 0xC475edbC00FFA10a75DBcdC29c6058005BB4cCa1
Mumbai: 
ERC-1155: Aa54A8E8BdEA1aa7E2ed7E5F681c798a8eD7e5AB
ERC-721: 0x0c702F78b889f25E3347fb978345F7eCF4F3861C
ERC-721: 0xd19018f7946D518D316BB10FdFF118C28835cF7a
WETH(ERC-20): 0xa6fa4fb5f76172d178d61b04b0ecd319c5d1c0aa 
Uniswap V3 Router (Undefined contract): 0x68b3465833fb72a70ecdf485e0e4c7bd8665fc45

File: packages/sysweb3-utils/src/contracts.ts [hunk 3]
```diff
@@ -24,6 +30,51 @@ export const isContractAddress = async (address: string, chainId = 1) => {
   return false;
 };
 
+export const contractChecker = async (
+  contractAddress: string,
+  network: INetwork
+) => {
+  setActiveNetwork(network);
+
+  const validateContractAddress = isContractAddress(contractAddress);
```
[the remark this answers is not in the document]
Comment (leozera14) [reply packages/sysweb3-utils/src/contracts.ts]: Fixed

File: packages/sysweb3-utils/src/contracts.ts [hunk 4]
```diff
@@ -24,6 +30,51 @@ export const isContractAddress = async (address: string, chainId = 1) => {
   return false;
 };
 
+export const contractChecker = async (
+  contractAddress: string,
+  network: INetwork
+) => {
+  setActiveNetwork(network);
```
[the remark this answers is not in the document]
Comment (leozera14) [reply packages/sysweb3-utils/src/contracts.ts]: Fixed

File: packages/sysweb3-utils/src/getContract.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    { abi: abi721, interfaceId: '0x80ac58cd', type: 'ERC-721' },
+    { abi: abi1155, interfaceId: '0xd9b67a26', type: 'ERC-1155' },
```
[the remark this answers is not in the document]
Comment (leozera14) [reply packages/sysweb3-utils/src/getContract.ts]: Fixed

File: packages/sysweb3-utils/src/getContractInfo.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import { Contract } from 'ethers';
+
+export const getContractInfo = async (contract: Contract) => {
```
[the remark this answers is not in the document]
Comment (leozera14) [reply packages/sysweb3-utils/src/getContractInfo.ts]: Fixed

File: packages/sysweb3-utils/src/contracts.ts [hunk 7]
```diff
@@ -24,6 +30,51 @@ export const isContractAddress = async (address: string, chainId = 1) => {
...
+
+    try {
+      const contractData = await getContract(contractAddress, network.url);
+
+      contract = contractData.contract as Contract | null;
+    } catch (_error) {
+      throw new Error(_error);
+    }
+
+    const contractInfos: any = await getContractInfo(contract as Contract);
```
[the remark this answers is not in the document]
Comment (leozera14) [reply packages/sysweb3-utils/src/contracts.ts]: Fixed

File: packages/sysweb3-utils/src/contractErrors.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
+export const CONTRACT_ERRORS: any = {
+  MAX_BATCH_MINT:
```
[the remark this answers is not in the document]
Comment (leozera14) [reply packages/sysweb3-utils/src/contractErrors.ts]: Fixed

File: packages/sysweb3-utils/src/getContract.ts (hunk 1 above)
Comment (leozera14) [reply packages/sysweb3-utils/src/getContract.ts]: Fixed

File: packages/sysweb3-utils/src/getContract.ts (hunk 2 above)
Comment (leozera14) [reply packages/sysweb3-utils/src/getContract.ts:34]: Fixed

**Review by claudiovb — APPROVED** [review]

**Pull request merged**